### PR TITLE
LPS-182297 | Test Automation Creation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-182297 - Product QA | Test Automation Creation

Hi! :)
This pr is corresponding to this ticket: [LPS-177031](https://issues.liferay.com/browse/LPS-177031)  → [LPS-182297](https://issues.liferay.com/browse/LPS-182297)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 1]
```diff
@@ -1392,6 +1392,249 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-177031. Page option Copy Page with Permissions works in content page and widget page for organization role"
+	@priority = 3
+	test ViewCopyPageWithPermissionsForOrganizationRole {
+		task ("Given a content page and a widget page with view permissions for Organization User") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1398]: new organization role and new regular role

**Review by Tim-Cao — COMMENTED** [review]
Hi @rafaellajordao, would you mind modifying according to comments? Thanks

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 2]
```diff
@@ -1782,6 +1782,262 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-177031. Page option Copy Page with Permissions works in content page and widget page for default roles"
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1785]: Sorry my mistake, Page option Copy Page with Permissions works in content page and widget page for available roles to normal site

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 3]
```diff
@@ -1782,6 +1782,262 @@ definition {
...
+			}
+
+			Role.viewPermissionPG(
+				permissionDefinitionKey = "CONTENT_PERMISSIONS_VIEW_CHECKBOX",
+				permissionFlagged = "true",
+				roleTitle = ${randomRoleKey});
+		}
+	}
+
+	@description = "This is a test for LPS-177031. Page option Copy Page with Permissions works in content page and widget page for new roles"
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1930]: Page option Copy Page with Permissions works in content page and widget page for available roles to organization site

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 4]
```diff
@@ -1782,6 +1782,262 @@ definition {
...
+					permissionFlagged = "false",
+					roleTitle = ${roleTitle});
+			}
+
+			Role.viewPermissionPG(
+				permissionDefinitionKey = "CONTENT_PERMISSIONS_VIEW_CHECKBOX",
+				permissionFlagged = "true",
+				roleTitle = ${randomRoleKey});
+
+			Button.clickSave();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1868]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 5]
```diff
@@ -1782,6 +1782,262 @@ definition {
...
+				menuItem = "Copy",
+				pageName = "Test Widget Page Name");
+
+			DropdownMenuItem.click(menuItem = "Page With Permissions");
+
+			SelectFrame(locator1 = "IFrame#MODAL_BODY");
+
+			PortletEntry.inputName(name = "Copy Widget Page Name");
+
+			Button.clickAdd();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1910]: Button.click(button = "Add");

			SelectFrame.selectFrameTop();

			Alert.viewSuccessMessage();

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 6]
```diff
@@ -1782,6 +1782,262 @@ definition {
...
+				menuItem = "Copy",
+				pageName = "Test Content Page Name");
+
+			DropdownMenuItem.click(menuItem = "Page With Permissions");
+
+			SelectFrame(locator1 = "IFrame#MODAL_BODY");
+
+			PortletEntry.inputName(name = "Copy Content Page Name");
+
+			Button.clickAdd();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1979]: Button.click(button = "Add");

			SelectFrame.selectFrameTop();

			Alert.viewSuccessMessage();

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 7]
```diff
@@ -1782,6 +1782,262 @@ definition {
...
+			PermissionsInline.gotoPagePG(
+				pageName = "Copy Content Page Name",
+				site = ${organizationName});
+
+			Role.viewPermissionPG(
+				permissionDefinitionKey = "CONTENT_PERMISSIONS_VIEW_CHECKBOX",
+				permissionFlagged = "true",
+				roleTitle = "Organization User");
+
+			Button.clickSave();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1994]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 8]
```diff
@@ -1782,6 +1782,262 @@ definition {
...
+				menuItem = "Copy",
+				pageName = "Test Widget Page Name");
+
+			DropdownMenuItem.click(menuItem = "Page With Permissions");
+
+			SelectFrame(locator1 = "IFrame#MODAL_BODY");
+
+			PortletEntry.inputName(name = "Copy Widget Page Name");
+
+			Button.clickAdd();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:2028]: Button.click(button = "Add");

			SelectFrame.selectFrameTop();

			Alert.viewSuccessMessage();

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 9]
```diff
@@ -1782,6 +1782,270 @@ definition {
...
+			}
+
+			Role.viewPermissionPG(
+				permissionDefinitionKey = "CONTENT_PERMISSIONS_VIEW_CHECKBOX",
+				permissionFlagged = "true",
+				roleTitle = ${randomRoleKey});
+		}
+
+		task ("Then the permissions must be the same as on the original pages") {
+			PagesAdmin.openPagesAdmin(siteURLKey = ${randomSiteName});
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1870]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 10]
```diff
@@ -1782,6 +1782,270 @@ definition {
...
+				site = ${organizationName});
+
+			Role.viewPermissionPG(
+				permissionDefinitionKey = "CONTENT_PERMISSIONS_VIEW_CHECKBOX",
+				permissionFlagged = "true",
+				roleTitle = "Organization User");
+		}
+
+		task ("Then the permissions must be the same as on the original pages") {
+			PagesAdmin.openPagesAdmin(siteURLKey = ${organizationName});
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:2002]: No need

**Comment by Tim-Cao** [comment]
See changes https://github.com/liferay-echo/liferay-portal/pull/12145
@rafaellajordao

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**